### PR TITLE
Cards paths powers tabs (part 1 - all form, no function)

### PR DESCRIPTION
**Pull request by alexmiddeleer**
I left the tabs as real buttons since it's fairly a11y friendly I'm assuming (that's why there's an orange outline if anyone's wondering, easy keyboard nav). If we think we need some of the fancier aria attributes I'm fine to put those in, but I don't know how world-changing they actually are.

### CODE REVIEW

File: next/components/tab-group.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { ThemeContext } from './theme-context';
+
+function TabGroup({ onChange, name, labels }) {
+  const [selectedLabel, setLabel] = useState(labels[0]);
+
+  useEffect(() => onChange(selectedLabel));
```
Comment (lily-pytel) [inline next/components/tab-group.js]: Placeholders for future functionality?

File: next/components/tab-group.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <ThemeContext.Consumer>
+      {theme => (
+        <div className="mg-tab-group" data-cy={`tabs_${name}`}>
+          <style jsx>{`
+            .mg-tab-group {
+              display: flex;
+              flex-wrap: nowrap;
+              font-size: 11px;
```
Comment (lily-pytel) [inline next/components/tab-group.js]: The mock has the font a little bigger (13.3px though I would just go with 14px): https://mythgard-hub.github.io/deckb_template.html

**Comment by lily-pytel** [comment]
Though I admit I hate that orange outline when you click it.

File: next/components/tab-group.js (hunk 1 above)
Comment (alexmiddeleer) [reply next/components/tab-group.js]: yep

File: next/components/tab-group.js (hunk 2 above)
Comment (alexmiddeleer) [reply next/components/tab-group.js]: For me it shows as 11px when I inspect it in Chrome..

**Comment by alexmiddeleer** [comment]
> Though I admit I hate that orange outline when you click it.

I don't think that part is great either. Open to suggestions.

File: next/components/tab-group.js (hunk 2 above)
Comment (lily-pytel) [reply next/components/tab-group.js]: That's so weird! I do think 11px is kind of small in general but it's obviously small potatoes

**Comment by lily-pytel** [comment]
> I don't think that part is great either. Open to suggestions.

How about just making it divs?

**Comment by alexmiddeleer** [comment]
> > I don't think that part is great either. Open to suggestions.
> 
> How about just making it divs?

It would still need some sort of outline and focus-ability, otherwise it can't be tab-navigated which is not accessible design. I think I'll remove the orange outline though and just let Chrome leave in its standard, nearly unnoticeable blue halo.

**Comment by jtrussell** [comment]
Another option would be to use links for tabs instead of buttons so each tab would get its own route. Added benefit: we could link directly to the paths/powers view of a decklist.

This might also push us toward tracking deck modifications in local storage like Lily was getting at [here](https://trello.com/c/wCR2rE1O/125-deck-builder-save-deck-when-user-is-not-logged-in#comment-5d6c7dff2b03901f6310fb54).

**Comment by lily-pytel** [comment]
I am super in favor of each tab getting its own route in general!

**Comment by jtrussell** [comment]
Agreed! The only awkward thing is Next doesn't give us a concept of "sub" routes out of the box (at least not that I'm aware of).

We'd have to do something like:

```
(Folders)
next/
  pages/
    deck-builder/
      index.js
      cards.js
      paths.js
      powers.js
```

Where `deck-builder/index.js` might render everything but the tab content and `deck-builder/paths.js` would do something like:

```javascript
import BuilderShell from './index'

export default function PathTab() {
  return (
    <BuilderShell>
      Path Tab!
    </BuilderShell>
  )
}
```

**Comment by jtrussell** [comment]
Actually, I take that back. It looks like there is some fancy dynamic routing stuff we could use!

https://github.com/zeit/next.js#dynamic-routing

**Comment by alexmiddeleer** [comment]
Oh wow, I didn’t even think of making the tabs change the url. Now I want
to make all of the filters part of the url 🤭.

**Comment by jtrussell** [comment]
I like it! Though don't feel compelled to overhaul before with v1 bearing down ;).

**Pull request merged**